### PR TITLE
Capnsquirrel/herder prototype

### PR DESCRIPTION
**Pull request by CapnSquirrel**
This PR contains no code related to the herder at all, sorry.
- Playable Architect and Chemist class
- Reworked Damageable Enemy and Player classes to work over network
- Various other changes to make game playable online

### CODE REVIEW

File: unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs [hunk 1]
```diff
@@ -172,15 +174,14 @@ void AttackPlayer(Collider other)
...
+        if (player.currentHealth <= 0)
         {
             FindNewTarget();
         }
     }
 
     void AttackZombie(Collider other)
     {
-        if (!isServer) return;
         var zombie = other.gameObject.GetComponent<DamagableEnemy>();
```
Comment (im-deepfriedwater) [inline unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs:187]: Should add back in.

**Review by im-deepfriedwater — CHANGES_REQUESTED** [review]
just needs that if statement back in, otherwise lgtm

File: unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs (hunk 1 above)
Comment (CapnSquirrel) [reply unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs:187]: ok, added back into the places where I thought appropriate. let me know.

Comment (im-deepfriedwater) [reply unity-ktwo/Assets/Scripts/Controllers/EnemyController.cs:187]: looks fine to me

**Review by im-deepfriedwater — APPROVED** [review]
changes in, approving.

**Pull request merged**